### PR TITLE
allow log to be passed for rabbit. update joi def

### PR DESCRIPTION
**Pull request by anandkumarpatel**
* allow log to be passed for rabbit model
* update joi def for schema

### CODE REVIEW

File: src/worker.js [hunk 1]
```diff
@@ -22,9 +22,7 @@ const optsSchema = joi.object({
   done: joi.func().required(),
   errorCat: joi.object(),
   finalRetryFn: joi.func(),
-  jobSchema: joi.object({
-    isJoi: joi.bool().valid(true)
-  }).unknown(),
+  jobSchema: joi.object().schema(),
```
Comment (thejsj) [inline src/worker.js]: 👍

File: src/rabbitmq.js [hunk 2]
```diff
@@ -57,19 +57,15 @@ class RabbitMQ {
     this.port = opts.port || parseInt(process.env.RABBITMQ_PORT, 10) || 5672
     this.username = opts.username || process.env.RABBITMQ_USERNAME
     this.password = opts.password || process.env.RABBITMQ_PASSWORD
-    this.log = logger.child({
-      module: 'ponos:rabbitmq',
-      hostname: this.hostname,
-      port: this.port,
-      clientName: this.name
-    })
+    this.log = opts.log || logger.child({ module: 'ponos:rabbitmq' })
```
Comment (thejsj) [inline src/rabbitmq.js]: So we were passing a logger before as a parameter (`ops.log`) and never using it?

Comment (anandkumarpatel) [inline src/rabbitmq.js]: this is for the rabbitMQ module specifically (used as publisher) but ponos server uses the logger we passed in

**Comment by thejsj** [comment]
LGTM

**Pull request merged**